File: solutions/0014-longest-common-prefix/solution.java

class Solution {
    public String longestCommonPrefix(String[] strs) {

        if (strs.length == 0) {
            return "";
        }

        // Sort the array
        Arrays.sort(strs);

        String ans = "";

        char[] first = strs[0].toCharArray();
        char[] last = strs[strs.length - 1].toCharArray();

        for (int i = 0; i < first.length; i++) {
            if (first[i] != last[i]) {
                break;
            }
            ans += first[i];
        }

        return ans;
    }
}

File: solutions/0014-longest-common-prefix/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {

        string s = "";

        sort(strs.begin(), strs.end());
        int n=strs.size();

        string first = strs[0];
        string second = strs[n - 1];

        for (int k = 0; k < first.size(); k++) {
            if (first[k] != second[k])
                break;
            s += first[k];
        }

        return s;
    }
};
